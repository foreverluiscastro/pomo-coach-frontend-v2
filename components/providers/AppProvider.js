import React, { useState, createContext, useContext } from "react";

const AppContext = createContext();

const AppProvider = ({ children }) => {
  const [user, setUser] = useState(null);

  // isStudy and session controls the app's color theme
  const [isStudy, setIsStudy] = useState(true);
  const [session, setSession] = useState("Study");

  // do not display the settings until the user clicks settings
  const [showSettings, setShowSettings] = useState(false);
  // do not display the login until the user clicks on the button
  const [showLogin, setShowLogin] = useState(false);

  // default values for time are 25 and 5 but can be customized
  const [studyTime, setStudyTime] = useState(25);
  const [breakTime, setBreakTime] = useState(5);

  // used to customize the study and break times
  const [settingsSaved, setSettingSaved] = useState(false);

  // for the start and pause functionality
  const [isRunning, setIsRunning] = useState(false);

  const [soundVolume, setSoundVolume] = useState(0.5);

  const appContextValue = {
    user,
    setUser,
    isStudy,
    setIsStudy,
    showSettings,
    setShowSettings,
    showLogin,
    setShowLogin,
    session,
    setSession,
    studyTime,
    setStudyTime,
    breakTime,
    setBreakTime,
    settingsSaved,
    setSettingSaved,
    isRunning,
    setIsRunning,
    soundVolume,
    setSoundVolume,
  };

  return (
    <AppContext.Provider value={appContextValue}>
      {children}
    </AppContext.Provider>
  );
};

export const useAppContext = () => useContext(AppContext);

export { AppProvider };
