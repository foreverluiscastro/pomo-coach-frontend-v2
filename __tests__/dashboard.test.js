import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import Dashboard from '@/pages/dashboard';
import { AppProvider } from '@/components/providers/AppProvider';
import { useRouter } from 'next/router';

jest.mock('next/router', () => ({
  useRouter: jest.fn(),
}));

describe('Dashboard', () => {
  beforeEach(() => {
    useRouter.mockReturnValue({
      push: jest.fn(),
      route: '/',
      pathname: '',
      query: {},
      asPath: '',
    });
  });


  it('renders the Study Stats', () => {
    render(<AppProvider><Dashboard/></AppProvider>)

    const heading = screen.getByRole('heading', { level: 1 })

    expect(heading).toBeInTheDocument()
  })

  it('renders Dashboard unchanged', () => {
    const { container } = render(<AppProvider><Dashboard/></AppProvider>)
    expect(container).toMatchSnapshot()
  })
})