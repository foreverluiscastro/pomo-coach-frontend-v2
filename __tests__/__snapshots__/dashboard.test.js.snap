// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard renders Dashboard unchanged 1`] = `
<div>
  <h1>
    Loading...
  </h1>
</div>
`;
